dali/operators/reader/loader/webdataset/tar_utils_test.cc

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/webdataset/tar_utils.h"
+#include <fcntl.h>
+#include <gtest/gtest.h>
+#include <libtar.h>
+#include <algorithm>
+#include <future>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "dali/operators/reader/loader/filesystem.h"
+#include "dali/util/std_file.h"
+
+namespace dali {
+namespace detail {
+
+TEST(LibTarUtilsTestSimple, Interface) {
+  std::string filepath(dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                                   "db/webdataset/MNIST/devel-1.tar"));
+  TarArchive archive(TarArchive(TarArchive(FileStream::Open(filepath, false, false))));
```

Comment:
If the purpose of this is to test move constructor I think it should be tested in separate test for better readability and in case of issues easier debugging.
Also I don't see the added value of multiple nested constructor calls in that context.

Reply:
Yes, that is what the whole tests is about. The subsequent reads just test if the moving was successful.